providers/implementations/signature/dsa_sig.c

```diff
@@ -131,11 +131,11 @@ static int dsa_setup_md(PROV_DSA_CTX *ctx,
                                                             sha1_allowed);
         size_t mdname_len = strlen(mdname);
 
-        if (md == NULL || md_nid == NID_undef) {
+        if (md == NULL || md_nid < 0) {
             if (md == NULL)
                 ERR_raise_data(ERR_LIB_PROV, PROV_R_INVALID_DIGEST,
                                "%s could not be fetched", mdname);
-            if (md_nid == NID_undef)
```

Comment:
Should this be `<=` rather than `<` ?

Reply:
This just makes it the same as ECDSA. I do not insist on that though as DSA is something legacy that should not be used with potential new digests anyway.